Kuroba/app/src/main/java/com/github/k1rakishou/chan/core/site/sites/wired7/Wired7Api.kt

```diff
@@ -0,0 +1,351 @@
+import com.github.k1rakishou.chan.core.site.parser.ChanReader
+import com.github.k1rakishou.chan.core.site.parser.processor.AbstractChanReaderProcessor
+import com.github.k1rakishou.chan.core.site.parser.processor.ChanReaderProcessor
+import com.github.k1rakishou.common.ModularResult
+import com.github.k1rakishou.common.isNotNullNorEmpty
+import com.github.k1rakishou.common.mutableListWithCap
+import com.github.k1rakishou.core_logger.Logger
+import com.github.k1rakishou.model.data.board.ChanBoard
+import com.github.k1rakishou.model.data.bookmark.ThreadBookmarkInfoObject
+import com.github.k1rakishou.model.data.bookmark.ThreadBookmarkInfoPostObject
+import com.github.k1rakishou.model.data.descriptor.BoardDescriptor
+import com.github.k1rakishou.model.data.descriptor.ChanDescriptor
+import com.github.k1rakishou.model.data.filter.FilterWatchCatalogInfoObject
+import com.github.k1rakishou.model.data.filter.FilterWatchCatalogThreadInfoObject
+import com.github.k1rakishou.model.data.post.ChanPostBuilder
+import com.github.k1rakishou.model.data.post.ChanPostHttpIcon
+import com.github.k1rakishou.model.data.post.ChanPostImage
+import com.github.k1rakishou.model.data.post.ChanPostImageBuilder
+import com.google.gson.stream.JsonReader
+import org.jsoup.parser.Parser
+import java.io.IOException
+import java.io.InputStream
+import java.io.InputStreamReader
+import java.util.*
+import kotlin.math.max
+
+@Suppress("BlockingMethodInNonBlockingContext")
+class Wired7Api(
+  private val siteManager: SiteManager,
+  private val boardManager: BoardManager,
+  commonSite: CommonSite
+) : CommonApi(commonSite) {
+
+  @Throws(Exception::class)
+  override suspend fun loadThreadFresh(
+    requestUrl: String,
+    responseBodyStream: InputStream,
+    chanReaderProcessor: ChanReaderProcessor
+  ) {
+    readBodyJson(responseBodyStream) { jsonReader ->
+      vichanReaderExtensions.iteratePostsInThread(jsonReader) { reader ->
+        readPostObject(reader, chanReaderProcessor)
+      }
+
+      chanReaderProcessor.applyChanReadOptions()
+    }
+  }
+
+  @Throws(Exception::class)
+  override suspend fun loadCatalog(
+    requestUrl: String,
+    responseBodyStream: InputStream,
+    chanReaderProcessor: AbstractChanReaderProcessor
+  ) {
+    readBodyJson(responseBodyStream) { jsonReader ->
+      vichanReaderExtensions.iterateThreadsInCatalog(jsonReader) { reader ->
+        readPostObject(reader, chanReaderProcessor)
+      }
+    }
+  }
+
+  @Throws(Exception::class)
+  private suspend fun readPostObject(reader: JsonReader, chanReaderProcessor: AbstractChanReaderProcessor) {
+    val builder = ChanPostBuilder()
+    builder.boardDescriptor(chanReaderProcessor.chanDescriptor.boardDescriptor())
+
+    val site = siteManager.bySiteDescriptor(chanReaderProcessor.chanDescriptor.siteDescriptor())
+      ?: return
+    val board = boardManager.byBoardDescriptor(chanReaderProcessor.chanDescriptor.boardDescriptor())
```

Comment:
You can probably do the same thing as here https://github.com/K1rakishou/Kuroba-Experimental/blob/b981bc96bf7b94653cda1dc1191817d12e5d5bfa/Kuroba/app/src/main/java/com/github/k1rakishou/chan/core/site/common/FutabaChanReader.kt#L103 which will allow users opening site boards without even loading them first.

Reply:
Fixed in [b4bfc0c](https://github.com/K1rakishou/Kuroba-Experimental/pull/750/commits/b4bfc0ca420ca71230f00819c457aa6c6b177461)